presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -607,4 +594,22 @@ private static ColumnMetadata hiddenColumn(String name, Type type)
     {
         return new ColumnMetadata(name, type, false, null, true);
     }
+
+    private void setTransactionId(long transactionId)
+    {
+        checkState(currentTransactionId.compareAndSet(null, transactionId), "current transaction ID already set");
+    }
+
+    private void clearRollback()
```

Comment:
why do we need to do this? the current metadata can only be used for the current transaction